### ISSUE TITLE
give useful error / warning message when device not connected to the internet

### ISSUE THREAD

**Issue by sashei** [mozilla-lockwise/lockwise-ios#415]
right now, we just quietly don't load anything :(

Summary of quick online/offline/online toggle testing from @devinreams:

1. blank white webviews do not contain useful "offline" errors or messages

2. depending on connectivity loss timing: a user could get really stuck after confirmation in "no entries" and not see their logins load in even after connectivity restored

### Visual Design

**InVision**
WebViews: https://mozilla.invisionapp.com/share/BQEUSO0S2#/311856390_No_Internet_-_WebView
FxA: https://mozilla.invisionapp.com/share/BQEUSO0S2#/311856391_No_Internet_-_FxA
Unpopulated Entry List: https://mozilla.invisionapp.com/share/BQEUSO0S2#/311856389_No_Internet_-_Entry_List_-_Not_Populated
Populated Entry List: https://mozilla.invisionapp.com/share/BQEUSO0S2#/311856388_No_Internet_-_Entry_List_-_Populated

**Zeplin**
WebViews: https://zpl.io/agvo46M
FxA: https://zpl.io/VOkMwA1
Unpopulated Entry List: https://zpl.io/bLO8l1G
Populated Entry List: https://zpl.io/Vk5NRWM

### Engineering Checklist
- [x] ItemList
- [x] WebView
- [x] FxA
- [x] unlock works on no network

**Comment by devinreams** [comment]
tldr:

1. blank white webviews do not contain useful "offline" errors or messages

2. depending on connectivity loss timing: a user could get really stuck after confirmation in "no entries" and not see their logins load in even after connectivity restored

@sandysage do you want to help consider and prioritize these?

@sashei @jhugman @sandysage do you think either are P1 must-address? I'm less worried about 1 but do worry a bit about 2. I'm not sure how feasible or complicated either solution(s) would be, though...

---

- ⚠️ Welcome, Get Started: no title just a close button with blank white screen
- ⚠️ Welcome, Learn More: "Learn More" title with blank white screen
- FxA, Sign In: depressed blue "Sign In" button, red "Working..." banner, then "Unexpected error" :ok:
- Confirm Account: the message just remains in place... and then disappears shortly after connection restored :ok:
- Populated List View: just shows what's already been loaded :ok:
-  🚨 Unpopulated List View: "syncing" message appears and disappears, "No entries found" blurb (misleading, incorrect) is shown leading user to possibly click a link that won't load for them...
-  ❌ Unpopulated Pull to Refresh: impossible to do so relies on #445, user appears to have to quit app to get first data loaded in?
- Populated Pull to Refresh: "syncing" message appears and disappears immediately, spinner doesn't appear and snaps back immediately, it just kinda silently fails...
- Item Detail, tap Web Address: opens in browser with connection error message :ok:
- ⚠️ Send Feedback / FAQ / Learn More: title and a blank white screen
- Settings, Account, Disconnect: works as expected :ok:

**Comment by jhugman** [comment]
This list is really helpful!

Of the concerns here: 

 * a minimal "No Internet Connection / Try later!" toast is needed for the Welcome / Login screen.
 * a similar toast would also be useful, but of lower priority for Pull to refresh.

We should try and avoid the app ending in a broken looking state, so we should detect  that no login screen is displayable, and then back out to the welcome screen.

For pull to refresh, we should remain on the list screen after the error has been displayed.

For bonus points, we should consider using the Reachability API and trigger a sync once we can see the network.

**Comment by devinreams** [comment]
>a minimal "No Internet Connection / Try later!" toast is needed for the Welcome / Login screen.
>a similar toast would also be useful, but of lower priority for Pull to refresh.

Agreed. Thanks for looking this over @jhugman.

This is prioritized as a P2 and in the inbox for when the time comes...
